src/HospitalLibrary/Patients/PatientService.cs

```diff
@@ -1,28 +1,43 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Entity.Core;
+using System.Data.SqlClient;
 using System.Threading.Tasks;
 using HospitalLibrary.Patients.Interfaces;
+using HospitalLibrary.Shared.Exceptions;
 using HospitalLibrary.Shared.Interfaces;
+using HospitalLibrary.Shared.Validators;
 
 namespace HospitalLibrary.Patients
 {
     public class PatientService: IPatientService
     {
-        private IUnitOfWork _unitOfWork;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IPatientRegistrationValidationService _registrationValidation;
 
-        public PatientService(IUnitOfWork unitOfWork)
+        public PatientService(IUnitOfWork unitOfWork,IPatientRegistrationValidationService registrationValidation)
         {
             _unitOfWork = unitOfWork;
+            _registrationValidation = registrationValidation;
         }
         
         public Task<IEnumerable<Patient>> GetAll()
         {
             return _unitOfWork.PatientRepository.GetAll();
         }
         
-        public Patient Create(Patient patient)
+        public async Task<Patient> Create(Patient patient)
         {
-            _unitOfWork.PatientRepository.Add(patient);
-            _unitOfWork.PatientRepository.Save();
+            await _registrationValidation.ValidatePatient(patient);
+            try
+            {
+                _unitOfWork.PatientRepository.Add(patient);
+                _unitOfWork.PatientRepository.Save();
+            }
+            catch(Exception  ex)
+            {
```

Comment:
Remove catch is you want to keep validator throwing exception. Otherwise, let database throw exception and leave it like this.